libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/StoreKeyConfig.java

```diff
@@ -122,14 +141,26 @@ public X509ExtendedKeyManager createKeyManager() {
 
     private X509ExtendedKeyManager createKeyManager(Path path) {
         try {
-            final KeyStore keyStore = readKeyStore(path);
+            KeyStore keyStore = readKeyStore(path);
+            keyStore = processKeyStore(keyStore);
             checkKeyStore(keyStore, path);
             return KeyStoreUtil.createKeyManager(keyStore, keyPassword, algorithm);
         } catch (GeneralSecurityException e) {
             throw keystoreException(path, e);
         }
     }
 
+    private KeyStore processKeyStore(KeyStore keyStore) {
+        if (filter == null) {
+            return keyStore;
+        }
+        final KeyStore processedKeystore = filter.apply(keyStore);
+        if (processedKeystore != null) {
+            return processedKeystore;
+        }
+        return keyStore;
```

Comment:
Shouldn't we throw an exception here? This silent ignoring filtering seems not intuitive